server/fleet/app.go

```diff
@@ -324,20 +324,40 @@ func (s *MacOSSettings) FromMap(m map[string]interface{}) (map[string]bool, erro
 
 		vals, ok := v.([]interface{})
 		if v == nil || ok {
-			strs := make([]string, 0, len(vals))
+			csSpecs := make([]MDMProfileSpec, 0, len(vals))
 			for _, v := range vals {
-				str, ok := v.(string)
-				if !ok {
-					// error, must be a []string
+				if m, ok := v.(map[string]interface{}); ok {
+					var spec MDMProfileSpec
+					// extract the Path field
+					if path, ok := m["path"].(string); ok {
+						spec.Path = path
+					}
+
+					// extract the Labels field
+					//
+					// TODO: what's the behavior for labels? not provided
+					// means that the labels are not modified or that
+					// should be cleaned?
```

Comment:
Good point. I think they should be cleared if the profile's path is provided but not the labels (which plays nice with gitops-based workflow - the file wouldn't change if there's no change, and if the labels are removed but the path still there IMO it's reasonable to assume it is now a standard, non label-based profile).

@noahtalerman and @marko-lisica for confirmation

Reply:
@mna Just spoke with Noah, and we agree they should be cleared.